Source/BugsnagErrorTypes.h

```diff
@@ -0,0 +1,49 @@
+//
+//  BugsnagErrorTypes.h
+//  Bugsnag
+//
+//  Created by Jamie Lynch on 22/04/2020.
+//  Copyright © 2020 Bugsnag. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+
+@interface BugsnagErrorTypes : NSObject
+
+/**
+ * Determines whether Out of Memory events should be reported to bugsnag.
+ *
+ * This flag is true by default.
+ */
+@property BOOL OOMs;
+
+/**
+ * Determines whether NSExceptions should be reported to bugsnag.
+ *
+ * This flag is true by default.
+ */
+@property BOOL NSExceptions;
+
+/**
+ * Determines whether signals should be reported to bugsnag.
+ *
+ * This flag is true by default.
+ */
+@property BOOL signals;
+
+/**
+ * Determines whether C errors should be reported to bugsnag.
+ *
+ * This flag is true by default.
+ */
+@property BOOL C;
```

Comment:
Are we justified in calling this C rather than C++? It seems correct, but not sure why the KS constant is `BSG_KSCrashTypeCPPException`?

Reply:
Addressed by other comments